taichi/struct/snode_tree.h

```diff
@@ -15,30 +17,46 @@ namespace lang {
 class SNodeTree {
  public:
   constexpr static int kFirstID = 0;
+  Program *prog;
 
   /**
    * Constructor.
    *
    * @param id Id of the tree
    * @param root Root of the tree
    */
-  explicit SNodeTree(int id, std::unique_ptr<SNode> root, bool packed);
+  explicit SNodeTree(int id,
+                     std::unique_ptr<SNode> root,
+                     bool packed,
+                     Program *prog);
+
+  void destroy();
```

Comment:
Could you add this as a method in `Program` itself? When destroying a `SNodeTree`, we can just free its memory and remove the tree from the program.

Reply:
`Destoryed` flag is not necessary. We can maintain it in `SNodeTreeBufferManager`. But `c=fb.finalize()` returns a pointer to the `SNodeTree` and `c.destroy()' destroys it. Then `destroy` method is needed in `SNodeTree`.

Reply:
You could have the python`SNodeTree` invoke the CPP `Program::destroy_snode_tree()` :-)

Reply:
Oh, make sense.